src/Symfony/Component/OptionsResolver/OptionsResolver.php

```diff
@@ -526,11 +526,11 @@ public function addAllowedValues($option, $allowedValues)
         }
 
         if (!isset($this->allowedValues[$option])) {
-            $this->allowedValues[$option] = $allowedValues;
```

Comment:
If we make this change in `master`, which uses modern PHP versions, we could even simplify it more:

```diff
-    if (!isset($this->allowedValues[$option])) {
-        $this->allowedValues[$option] = $allowedValues;
-    } else {
-        $this->allowedValues[$option] = array_merge($this->allowedValues[$option], $allowedValues);
-    }

+    $this->allowedValues[$option] = array_merge($this->allowedValues[$option] ?? array(), $allowedValues);
```

Reply:
Those if/else constructs aren't useless: they prevent a unnecessary call to `array_merge`, which is a costly function, even though the damage should be limited when it is called with an empty array as argument.
As such, these changes don't do anything valuable, they just reduce the number of lines of code at the expense of performance, so they should be left untouched IMHO.

Reply:
@skalpa it is a really minor cost.

Reply:
a minor cost repeated hundreds or thousands times when building complex forms with many fields becomes major

Reply:
I agree with stof on this one. Perhaps it would be a good idea to add a comment as of _why_ this was done, instead of always calling array_merge?

Reply:
removed